src/librustc_session/session.rs

```diff
@@ -235,7 +235,37 @@ impl Session {
     }
 
     pub fn local_crate_disambiguator(&self) -> CrateDisambiguator {
-        *self.crate_disambiguator.get()
+        self.crate_disambiguator.get().copied().unwrap()
+    }
+
+    #[inline]
+    pub fn crate_types(&self) -> &[CrateType] {
+        self.crate_types_opt().unwrap()
+    }
+
+    #[inline]
+    pub fn crate_types_opt(&self) -> Option<&[CrateType]> {
```

Comment:
We can leave this to a future PR but it seems quite suspect if we're using this API, as that presumes we don't really care about the answer?

Reply:
It's only used here:

https://github.com/rust-lang/rust/blob/31add7e60709445617ab54a69f6f21cfcb2e3122/src/librustc_codegen_llvm/context.rs#L103

Reply:
I looked at that but wasn't able to quickly tell what all is using it and at what stage of compilation -- it does seem like crate types should be populated by then, but I could be wrong of course :)

Reply:
Sorry. I see I forgot to respond to this. I did initially try unwrapping `crate_types` here, but it wasn't populated at this point.

Reply:
No worries. Glad we at least tried it - it sounds like there may be something that's not quite working somewhere as a result, maybe good to file a follow-up issue. Feel free to assign to me. Certainly not urgent.

Reply:
Will do.